packages/cadl-ranch-specs/http/type/property/additional-properties/main.tsp

```diff
@@ -0,0 +1,79 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+import "@azure-tools/typespec-azure-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Illustrates various property types for models")
+@scenarioService("/type/property/additionalProperties")
+namespace Type.Property.AdditionalProperties;
+
+#suppress "@azure-tools/typespec-azure-core/bad-record-type" "For testing"
+@doc("The model extends from Record<unknown> type.")
+model ModelExtendsRecordUnknown extends Record<unknown> {
```

Comment:
@timotheeguerin I'd like to confirm my understanding here.

Any model which extends Record<> would enbale its `additionalProperties` flag, no matter `extends Record<string>` or `extends Record<float>`.

Reply:
well it would enable additional properties of the type provide in the record. I would not expect `is Record<string>` and `is Record<float>` to produce the same model.

I guess yeah we could add more types than just unknown. Similar to what we have for property type.

Reply:
I think Record<unknown> is wider than Record<string> and Record<float>, etc. I add the test for check int/string/boolean: https://github.com/Azure/cadl-ranch/blob/2f4816a30f06c18c049dbaf0c2a1e8cd9cf18f83/packages/cadl-ranch-specs/http/type/property/additional-properties/mockapi.ts#L44-L49
So there is no need to add Record<string/...> I think.

Reply:
@msyyc I think it can still be valuable to make sure you generate the right thing but we can come back to it

Reply:
> @timotheeguerin I'd like to confirm my understanding here:
> 
> Any model which extends Record<> would enbale its `additionalProperties` flag, no matter `extends Record<string>` or `extends Record<float>`.
> 
> If yes, do you think we need to also add other types cases?

I think any `Record<>` would enable the `additionalProperties` feature.
However the detail of the type/class in language would likely be a bit different depends on the type in Record.
E.g. Java would have `Map<String, String>` instead of `Map<String, Object>`, if type in Record is `string`.

Reply:
@weidongxu-microsoft @MaryGao I add `Record<string>` and `Record<float>`. But actually Python doesn't care about the type, so please feel free to commit directly if you need.

Reply:
nit, if we prefer service to use `is Record<>`, we can have less type coverage on `extends Record<>` (maybe only one `Record<unknown>` be enough for `extends`).

@chunyu3 @pshao25 Is .NET OK with this PR?